ironfish/src/consensus/verifier.ts

```diff
@@ -134,15 +135,13 @@ export class Verifier {
    *
    * @returns deserialized transaction to be processed by the main handler.
    */
-  verifyNewTransaction(serializedTransaction: SerializedTransaction): Transaction {
+  verifyNewTransaction(data: SerializedTransaction): Transaction {
     try {
-      const transaction = new Transaction(serializedTransaction)
+      const transaction = parseTransaction(data)
 
-      // Transaction is lazily deserialized, so we use takeReference()
+      // Transaction is lazily deserialized, so we use withReference()
       // to force deserialization errors here
-      transaction.takeReference()
-
-      return transaction
+      return transaction.withReference(() => transaction)
```

Comment:
Oh interesting.. I wonder if this has any implications (or if this _had_ unintended implications)